src/platform/unix/linux/mod.rs

```diff
@@ -26,18 +26,64 @@ use std::thread;
 
 const MAX_FDS_IN_CMSG: u32 = 64;
 
+const SCM_RIGHTS: c_int = 0x01;
+
+// The value Linux returns for SO_SNDBUF
+// is not the size we are actually allowed to use...
+// Empirically, we have to deduct 32 bytes from that.
+const RESERVED_SIZE: usize = 32;
+
+#[cfg(target_os="android")]
+const TEMP_FILE_TEMPLATE: &'static str = "/sdcard/servo/ipc-channel-shared-memory.XXXXXX";
+
+#[cfg(not(target_os="android"))]
+const TEMP_FILE_TEMPLATE: &'static str = "/tmp/ipc-channel-shared-memory.XXXXXX";
+
+
+#[cfg(target_os="linux")]
+type IovLen = usize;
+#[cfg(target_os="linux")]
+type SockLen = usize;
+#[cfg(target_os="linux")]
+#[inline]
+unsafe fn new_sockaddr_un(path: *const c_char) -> (sockaddr_un, usize) {
+    let mut sockaddr = sockaddr_un {
+        sun_family: libc::AF_UNIX as u16,
+        sun_path: [ 0; 108 ],
+    };
+    libc::strncpy(sockaddr.sun_path.as_mut_ptr(),
+                  path,
+                  sockaddr.sun_path.len() - 1);
+    let len = mem::size_of::<sockaddr_un>() - 104 + (libc::strlen(sockaddr.sun_path.as_ptr()) as usize);
```

Comment:
Yes, this is an error. `linux` would be `- 108`, while `freebsd` would be `- 104`